### PR TITLE
Post-processing trending FEE task

### PR DESCRIPTION
**Pull request by KotliarovAr**
The task works with 2 histograms: 1D "LaneStatusSummary" (IB, ML, OL) and 2D "TriggerVsFeeid". A specially designed reductor "ReductorBinContent" extracts the needed values used to plot graphs. Totally, the task produces 4 plots: "LaneStatusSummary" for all parts of the ITS and 1 plot for Triggers.

There are 2 possibilities for how the graphs can be represented: as a function of "Run number" or "Time". One needs to specify in the itsQCTrendingFEE.json "selection" criterion "Time or Run".

### CODE REVIEW

File: Modules/ITS/itsQCTrendingFEE.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        "dataSources": [
+          {
+            "type": "repository",
+            "paths": [ "ITS/MO/ITSFEE/LaneStatusSummary",
+                       "ITS/MO/ITSFEE/LaneStatusSummary",
+                       "ITS/MO/ITSFEE/LaneStatusSummary",
+                       "ITS/MO/ITSFEE" ],
+            "names": [ "LaneStatusSummaryIB",
+                       "LaneStatusSummaryML",
+                       "LaneStatusSummaryOL",
```
Comment (iravasen) [inline Modules/ITS/itsQCTrendingFEE.json:40]: Can we include also `LaneStatusSummaryGlobal` which contains the total amount of lanes in Error/Warning/Fault? I think this plot can go to the shifter's layout. Then we keep IB/ML/OL plots + trigger plots for experts

File: Modules/ITS/itsQCTrendingFEE.json [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                       "TriggerVsFeeid" ],
+            "reductorName": "o2::quality_control_modules::its::ReductorBinContent",
+            "moduleName": "QcITS"
+          }
+        ],
+        "plots": [
+          {
+            "names": [ "Number of \"Warnings\"" ],
+            "title": [ "Number of WARNINGS trend in IB" ],
```
Comment (iravasen) [inline Modules/ITS/itsQCTrendingFEE.json:49]: Can you please change the title in "IB - Trend of number of lanes in WARNING" (the same for the other plots)?

File: Modules/ITS/src/TrendingTaskITSFEE.cxx [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    }
+
+    // Retrieve data for trend plot
+    mTrend->Draw(plot.varexp.c_str(), "", "goff");
+    std::vector<double> dataRetrieve(mTrend->GetV1(), mTrend->GetV1() + mTrend->GetSelectedRows() % mTrend->GetEstimate());
+
+    // Initialize MultiGraph and Legend
+    if (countplots == 0) {
+      multi_trend = new TMultiGraph();
+      SetGraphName(multi_trend, plot.name, Form("Lane status summary trend %s", itsParts[countITSpart].c_str()));
```
Comment (iravasen) [inline Modules/ITS/src/TrendingTaskITSFEE.cxx:167]: can't you use `plot.title` for the third member?

File: Modules/ITS/src/TrendingTaskITSFEE.cxx [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+      // Draw plots
+      if (hDummy)
+        hDummy->Draw();
+      multi_trend->Draw(Form("%s", hDummy ? "" : "a"));
+      legend->Draw("SAME");
+
+      // Upload plots
+      auto mo = std::make_shared<MonitorObject>(canvas, mConfig.taskName, "o2::quality_control_modules::its::TrendingTaskITSFEE", mConfig.detectorName);
+      mo->setIsOwner(false);
+      qcdb.storeMO(mo);
```
Comment (iravasen) [inline Modules/ITS/src/TrendingTaskITSFEE.cxx:208]: If I understand correctly, here you store to ccdb every single plot as defined in the json. If so, can you remove this and keep only the `storeMO` related to the storage of the canvases with all trends together?

File: Modules/ITS/src/TrendingTaskITSFEE.cxx [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+  TGraph* trend_plot_Flags[nFlags];
+
+  // Retrieve X-axis for trend plot. Two options: "Time" or "Run"
+  long int numberOfEntries = mTrend->GetEntriesFast();
+  mTrend->Draw("time", "", "goff");
+  std::vector<double> retrieveTime(mTrend->GetV1(), mTrend->GetV1() + mTrend->GetSelectedRows() % mTrend->GetEstimate());
+
+  std::vector<double> retrieveRunNumber;
+  for (int j = 1; j <= numberOfEntries; j++)
+    retrieveRunNumber.push_back((double)j);
```
Comment (IsakovAD) [inline Modules/ITS/src/TrendingTaskITSFEE.cxx:146]: Would it be possible to unify construction of the x-axis? I would use as an example FHR task approach where we obtain X/Y axis with mTrend->GetV1() and mTrend->GetV2() function

File: Modules/ITS/src/TrendingTaskITSFEE.cxx [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+      multi_trend = new TMultiGraph();
+      SetGraphName(multi_trend, plot.name, Form("Lane status summary trend %s", itsParts[countITSpart].c_str()));
+
+      legend = new TLegend(0.93, 0.55, 1.0, 0.9);
+      SetLegendStyle(legend, Form("LaneStatusSummary_%s_legend", itsParts[countITSpart].c_str()));
+
+      isRun = plot.selection.find("Run") != std::string::npos ? true : false;
+    }
+
+    trend_plot_Flags[countplots] = new TGraph(numberOfEntries, isRun ? &retrieveRunNumber[0] : &retrieveTime[0], &dataRetrieve[0]);
```
Comment (IsakovAD) [inline Modules/ITS/src/TrendingTaskITSFEE.cxx:175]: Could you delete the **trend_plot_Flags** object?

**Comment by iravasen** [comment]
Thanks @KotliarovAr , approving.

**Pull request merged**